src/interfaces.ts

```diff
@@ -2,15 +2,23 @@ import * as React from 'react';
 export type ComponentInstance<P = any, S = any> = new () => React.Component<P, S>;
 
 export type Validator = (model: any, allProps?: any) => boolean;
+
 export type ErrorMessage = {
     [name: string]: string | undefined;
 };
 
 export type ValidationRule = <T = any>(
     prop: keyof T,
     errorMessage: string,
-    attr?: any
-) => [Validator, ErrorMessage];
+    type?: ValidationType
+) => ValidationRuleResult;
+
+export type ValidationRuleResult = [Validator, ErrorMessage, ValidationType];
+
+export enum ValidationType {
```

Comment:
I prefer `export const enum` vs `export enum` — see transpiled JS for both of these